File: app/src/main/java/androidsamples/java/dicegames/Die6.java
package androidsamples.java.dicegames;

import java.util.Random;

/**
 * An implementation of a six-faced {@link Die} using {@link Random}.
 */
public class Die6 implements Die {
  Random rng;
  int value;

  public Die6() {
    rng = new Random();
  }

  @Override
  public void roll() {
    value = 1 + rng.nextInt(6);
  }

  @Override
  public int value() {
    return value;
  }
}

File: app/src/main/java/androidsamples/java/dicegames/Die6.kt
package androidsamples.java.dicegames

import java.util.*

/**
 * An implementation of a six-faced [Die] using [Random].
 */
class Die6 : Die {
    var rng: Random
    var value = 0

    init {
        rng = Random()
    }

    override fun roll() {
        value = 1 + rng.nextInt(6)
    }

    override fun value(): Int {
        return value
    }
}